src/sentry/api/endpoints/organization_releases.py

```diff
@@ -161,6 +162,11 @@ def post(self, request, organization):
             if commit_list:
                 release.set_commits(commit_list)
 
+            head_commits = result.get('head_commits')
+            if head_commits:
+                fetch_commits = request.user.is_authenticated() and not commit_list
+                release.set_head_commits(head_commits, request.user, fetch_commits=fetch_commits)
```

Comment:
Can we not fetch lazy and retry? Also I can see us actually making release of stuff not yet on github.